database-commons/src/main/java/io/cdap/plugin/db/sink/AbstractDBSink.java

```diff
@@ -229,6 +241,9 @@ public void prepareRun(BatchSinkContext context) {
     }
 
     addOutputContext(context);
+
+    // set error details provider
+    context.setErrorDetailsProvider(new ErrorDetailsProviderSpec(getErrorDetailsProviderClassName()));
```

Comment:
`context.setErrorDetailsProvider` should be called before `addOutputContext(context)` is called.

Otherwise error details provider information is not propagated to platform.

Reply:
Understood, moved !